src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,225 @@
+                    + car + ". ", throwable);
+        }
+    }
+
+    private void removeAllDriversByCar(Car car) {
+        String query = "DELETE FROM cars_drivers WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement removeCarsDriversStatement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            removeCarsDriversStatement.setLong(1, car.getId());
+            removeCarsDriversStatement.executeUpdate();
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't remove drivers from car "
+                    + car + ". ", throwable);
+        }
+    }
+
+    private Set<Driver> getAllDriversByCarId(Long carId) {
+        String query = "SELECT cars.id AS car_id, "
+                + "drivers.id AS driver_id,  "
+                + "drivers.`name` AS driver_name,  "
+                + "drivers.license_number AS driver_license "
+                + "FROM cars "
+                + "JOIN cars_drivers "
+                + "ON cars_drivers.car_id = cars.id "
+                + "JOIN drivers "
+                + "ON cars_drivers.driver_id = drivers.id;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getDriversStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = getDriversStatement.executeQuery();
+            Set<Driver> drivers = new HashSet<>();
+            while (resultSet.next()) {
+                Long driverCarId = resultSet.getObject("car_id", Long.class);
+                Driver driver = parseDriverFromResultSet(resultSet);
+                if (carId.equals(driverCarId)) {
+                    drivers.add(driver);
+                }
+            }
+            return drivers;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get list of drivers for car with ID:"
+                    + carId + " !", throwable);
+        }
+    }
+
+    private Driver parseDriverFromResultSet(ResultSet resultSet) throws SQLException {
+        Long driverId = resultSet.getObject("driver_id", Long.class);
+        String driverName = resultSet.getString("driver_name");
+        String driverLicense = resultSet.getString("driver_license");
+        Driver driver = new Driver(driverName, driverLicense);
+        driver.setId(driverId);
+        return driver;
+    }
+
+    private Car parseCarFromResultSet(ResultSet resultSet) throws SQLException {
+        String manufacturerName = resultSet.getString("manufacturer_name");
+        Long manufacturerId = resultSet.getObject("manufacturers_id", Long.class);
+        String manufacturerCountry = resultSet.getString("country");
+        Manufacturer manufacturer = new Manufacturer(manufacturerName, manufacturerCountry);
+        manufacturer.setId(manufacturerId);
```

Comment:
lets move this code to separate method parseManufacturer FromResultSet

Reply:
Fixed.